src/valid/type.rs

```diff
@@ -239,9 +245,9 @@ impl super::Validator {
                 // `DATA`.
                 let base_info = &self.types[base.index()];
                 let data_flag = if base_info.flags.contains(TypeFlags::SIZED) {
-                    TypeFlags::DATA
+                    TypeFlags::DATA | TypeFlags::ARGUMENT
```

Comment:
shouldn't the `ARGUMENT` always be there for pointers? currently there is a branch of the code where it doesn't get set

Reply:
upon closer inspection this isn't true since the following wgsl is invalid (according to our testing suite)
```rust
fn unacceptable_unsized(arg: ptr<storage, array<f32>>) { }
```